flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NetworkEnvironment.java

```diff
@@ -150,6 +159,11 @@ public NetworkEnvironmentConfiguration getConfiguration() {
 		return config;
 	}
 
+	@VisibleForTesting
+	public Map<IntermediateDataSetID, SingleInputGate> getInputGatesById() {
```

Comment:
Do we need to expose the map or would it be enough to have `getInputGate(IntermediateDataSetID)`?

Reply:
We should actually inject it as dependency to check in tests